elastic/kibana
# [Uptime] Default alert connectors email settings

## Summary

Fixes https://github.com/elastic/kibana/issues/116362
Fixes https://github.com/elastic/kibana/issues/121098

Added ability to select email connectors in uptime alert connectors settings. 

When email connector is selected. Section to add default email appears. Where user can select, To, Cc, Bcc emails. 

Those emails will be used to send alerts via selected email connectors in the uptime page.

![image](https://user-images.githubusercontent.com/3505601/150306555-0b76fc57-0952-4450-9e25-166dce60f69c.png)


### Testing 

- Add an email connector using rule and management page or from the uptime setting page using Add connector button
- Configure email connector via https://nodemailer.com/app/ , using this app you can start a dummy email server and specify settings in the email connector like this

```
    host: 'localhost',
    port: 1025,
    auth: {
        user: 'project.1',
        pass: 'secret.1'
    }

```

- Once email connector is configure you can test it within the connector UI to make sure it works and sends an email, that email will appear in nodemailer app
- Select that connector in the uptime setting default alert connectors
- Fields to add `To`, `Cc`, `Bcc` should appear in the settings page 
- If any email connectors is selected `To` becomes a required field. 
- All email fields should check for valid emails

## Note
Unit testing this component was hard, so i opted for more streamlined e2e tests.
